src/toil/lib/ec2.py

```diff
@@ -513,10 +514,17 @@ def create_auto_scaling_group(autoscaling_client: BaseClient,
 
     if len(vpc_subnets) == 0:
         raise RuntimeError("No VPC subnets specified to launch into; not clear where to put instances")
+        
+    def get_launch_template_spec(instance_type):
+        """
+        Get a LaunchTemplateSpecification for the given instance type.
+        """
+        return {'LaunchTemplateId': launch_template_ids[instance_type], 'Version': '$Default'}
 
     # We always write the ASG with a MixedInstancesPolicy even when we have only one type.
-    mip = {'LaunchTemplate': {'LaunchTemplateSpecification': {'LaunchTemplateId': launch_template_id, 'Version': '$Default'},
-                              'Overrides': [{'InstanceType': t} for t in instance_types]}}
+    # And we use a separate launch template for every instance type, and apply it as an override.
```

Comment:
The only way to hook multiple instance types onto one autoscaling group is via overrides. And we're using additional templates so that node storage overrides can work. I will note that.